code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/EventHub.kt

```diff
@@ -182,7 +198,15 @@ internal class EventHub {
             )
         }
 
-        // TODO: Record event to event history database if required.
+        event.mask?.let {
```

Comment:
This should be moved to SerialWorkDispatcher.WorkHandler. We already have a Todo marked there for this logic. 
By keeping the logic here, we will miss recording events generated as result of dispatch consequence.